storage/innobase/xtrabackup/src/keyring_components.cc

```diff
@@ -214,5 +219,48 @@ bool keyring_init_offline() {
   keyring_component_initialized = true;
   return true;
 }
+
+static bool get_one_option(int optid MY_ATTRIBUTE((unused)),
+                           const struct my_option *opt MY_ATTRIBUTE((unused)),
+                           char *argument MY_ATTRIBUTE((unused))) {
+  return (FALSE);
+}
+
+static bool read_server_uuid() {
+  char *uuid = NULL;
+  MEM_ROOT argv_alloc{PSI_NOT_INSTRUMENTED, 512};
+  const char *groups[] = {"mysqld", NULL};
+
+  my_option config_options[] = {
+      {"server-uuid", 0, "", &uuid, &uuid, 0, GET_STR, REQUIRED_ARG, 0, 0, 0, 0,
+       0, 0},
+      {0, 0, 0, 0, 0, 0, GET_NO_ARG, NO_ARG, 0, 0, 0, 0, 0, 0}};
+
+  char *exename = (char *)"xtrabackup";
+  char **backup_my_argv = &exename;
+  int backup_my_argc = 1;
+  char fname[FN_REFLEN];
+
+  /* we need full name so that only backup-my.cnf will be read */
+  if (fn_format(fname, "backup-my.cnf", xtrabackup_target_dir, "",
+                MY_UNPACK_FILENAME | MY_SAFE_PATH) == NULL) {
+    return (false);
+  }
+
+  if (my_load_defaults(fname, groups, &backup_my_argc, &backup_my_argv,
+                       &argv_alloc, NULL)) {
+    return (false);
+  }
+
```

Comment:
can we refactor a bit. I see same code in load_backup_my_cnf() for a different option?